parser_library/src/workspaces/file_manager_impl.cpp

```diff
@@ -107,13 +106,14 @@ processor_file_ptr file_manager_impl::find_processor_file(const std::string& key
     return change_into_processor_file_if_not_already_(ret->second);
 }
 
-list_directory_result file_manager_impl::list_directory_files(const std::string& path)
+list_directory_result file_manager_impl::list_directory_files(const utils::path::resource_location& directory)
 {
-    std::filesystem::path lib_p(path);
+    std::filesystem::path lib_p(directory.get_path());
```

Comment:
add a simple check - if not path, return unable to open equivalent?

Reply:
The check is carried out when `utils::path::list_directory_regular_files()` is called a few lines below and it seems sufficient to me. Do you agree?

Reply:
When an URI is passed in, the `get_path` returns empty string, right? At that point, won't we start listing like the current directory, or something like that?

Reply:
Figured it should probably need to be relocated to the reading API so we can get directories from URIs. I'm thinking that the originally proposed check is not needed anymore. What do you think?